File: 1. Two Sum/242. Valid Anagram/Validanagram.java
class Validanagram {
  public boolean isAnagram(String s, String t) {
      int[] letterCount = new int[26];
      if(s.length() != t.length()) return false;
      for(int i = 0; i < s.length(); i++) {
          letterCount[s.charAt(i) - 'a']++;
      }
      for(int i = 0; i < t.length(); i++) {
          letterCount[t.charAt(i) - 'a']--;
      }
      for(int i = 0; i < 26; i++) {
          if(letterCount[i] != 0) return false;
      }
      return true;
  }
}

//Validanagram v = new Validanagram();
//System.out.println(v.isAnagram("tar", "rat"));

File: 1. Two Sum/242. Valid Anagram/validanagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        letterCount = [0] * 26
        if len(s) != len(t):
            return False
        for char in s:
            letterCount[ord(char) - ord('a')]+=1
        for char in t:
            letterCount[ord(char) - ord('a')]-=1
        for val in letterCount:
            if val != 0:
                return False
        return True

#p = Solution()
#print(p.isAnagram("tar", "rat"))